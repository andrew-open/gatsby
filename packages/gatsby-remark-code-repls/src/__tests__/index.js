jest.mock(`fs`, () => {
  return {
    existsSync: () => true,
    readFileSync: () => 'const foo = "bar";',
  }
})

const fs = require(`fs`)
const Remark = require(`remark`)
const plugin = require(`../`)

const remark = new Remark()

describe(`gatsby-remark-code-repls`, () => {
  describe('Babel REPL', () => {
    it(`generates a link for the specified example file`, async () => {
      const markdownAST = remark.parse(`[Babel](babel-repl://file.js)`)

      const transformed = plugin({ markdownAST }, { directory: 'example-directory' })

      expect(transformed).toMatchSnapshot()
    })

    it(`generates a link with the specified target`, async () => {
      const markdownAST = remark.parse(`[Babel](babel-repl://file.js)`)

      const transformed = plugin({ markdownAST }, { directory: 'example-directory', target: '_blank' })

      expect(transformed).toMatchSnapshot()
    })

    it(`generates a link for files in nested directories`, async () => {
      const markdownAST = remark.parse(`[Babel](babel-repl://path/to/nested/file.js)`)

      const transformed = plugin({ markdownAST }, { directory: 'example-directory' })

      expect(transformed).toMatchSnapshot()
    })

    it(`errors if provided a link to a local file that does not exist`, async () => {
      fs.existsSync = () => false

      const markdownAST = remark.parse(`[Babel](babel-repl://file.js)`)

      expect(() => {
        plugin({ markdownAST }, {});
      }).toThrow();
    })
  })

  // TODO Test Codepen links and such
})
