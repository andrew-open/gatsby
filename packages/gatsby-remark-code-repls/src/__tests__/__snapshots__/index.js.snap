// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-remark-code-repls Babel REPL generates a link for files in nested directories 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "html",
          "value": "<a href=\\"https://babeljs.io/repl/#?presets=react&code_lz=MYewdgzgLgBAZiEMC8MBEAjAhgJzQbiA\\" >",
        },
        Object {
          "type": "text",
          "value": "Babel",
        },
        Object {
          "type": "html",
          "value": "</a>",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 45,
          "line": 1,
          "offset": 44,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 45,
      "line": 1,
      "offset": 44,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`gatsby-remark-code-repls Babel REPL generates a link for the specified example file 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "html",
          "value": "<a href=\\"https://babeljs.io/repl/#?presets=react&code_lz=MYewdgzgLgBAZiEMC8MBEAjAhgJzQbiA\\" >",
        },
        Object {
          "type": "text",
          "value": "Babel",
        },
        Object {
          "type": "html",
          "value": "</a>",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 30,
          "line": 1,
          "offset": 29,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 30,
      "line": 1,
      "offset": 29,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`gatsby-remark-code-repls Babel REPL generates a link with the specified target 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "html",
          "value": "<a href=\\"https://babeljs.io/repl/#?presets=react&code_lz=MYewdgzgLgBAZiEMC8MBEAjAhgJzQbiA\\" target=\\"_blank\\">",
        },
        Object {
          "type": "text",
          "value": "Babel",
        },
        Object {
          "type": "html",
          "value": "</a>",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 30,
          "line": 1,
          "offset": 29,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 30,
      "line": 1,
      "offset": 29,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;
